pkg/kubectl/cmd/logs.go

```diff
@@ -233,14 +242,35 @@ func (o LogsOptions) RunLogs() error {
 	switch t := o.Object.(type) {
 	case *api.PodList:
 		for _, p := range t.Items {
-			if err := o.getLogs(&p); err != nil {
+			if err := o.getPodLogs(&p); err != nil {
 				return err
 			}
 		}
 		return nil
 	default:
-		return o.getLogs(o.Object)
+		return o.getPodLogs(o.Object)
+	}
+}
+
+// getpodLogs checks whether the obj is a pod and o.AllContainers is set to true.
+// If so, it retrives all containers' log in the pod.
+func (o LogsOptions) getPodLogs(obj runtime.Object) error {
+	if pod, ok := obj.(*api.Pod); ok && o.AllContainers {
```

Comment:
Since there's already a type switch where this method is being invoked, why not move the type check there? This will simplify this method to:
```go
if !o.AllContainers {
    return o.getLogs(pod)
}
for ...
for ...
return nil
```

Reply:
Done.